rotkehlchen/chain/ethereum/modules/convex/convex_cache.py

```diff
@@ -0,0 +1,146 @@
+            key_parts=(CacheType.CONVEX_POOL_ADDRESS,),
+            values=[pool.pool_address],
+        )
+        globaldb_set_unique_cache_value(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.CONVEX_POOL_NAME, pool.pool_address),
+            value=pool.pool_name,
+        )
+
+
+def read_convex_data_from_cache() -> tuple[dict[ChecksumEvmAddress, str]]:
+    """Reads convex pools and names from global db cache tables."""
+    pools: dict[ChecksumEvmAddress, str] = {}
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        for pool_address in get_existing_pools(cursor):
+            if (pool_name := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.CONVEX_POOL_NAME, pool_address),
+            )) is None:
+                continue
+            pools[pool_address] = pool_name
+
+    return (pools,)
+
+
+def query_convex_data_from_chain(
+        ethereum: 'EthereumInquirer',
+        existing_pools: set[ChecksumEvmAddress],
+) -> dict[ChecksumEvmAddress, str]:
+    """
+    Query Booster contract and fetch all reward pools. It creates a mapping for convex pools
+    to the symbol of the underlying lp token. It uses the `existing_pools` argument to ensure
+    that additional queries are not made for pools that are already cached.
+
+    May raise:
+    - RemoteError if failed to query chain
+    """
+    booster_contract = ethereum.contracts.contract(BOOSTER)
+    pools_count = booster_contract.call(node_inquirer=ethereum, method_name='poolLength')
+    calls_to_booster = [(
+        booster_contract.address,
+        booster_contract.encode('poolInfo', [x]),
+    ) for x in range(pools_count)]
+    booster_result = ethereum.multicall(calls=calls_to_booster)
+    convex_rewards_addrs: list[ChecksumEvmAddress] = []
+    convex_lp_tokens_addrs = []
+    for single_booster_result in booster_result:
+        crv_rewards = hex_or_bytes_to_address(single_booster_result[96:128])
+        if crv_rewards not in existing_pools:
+            lp_token_addr = hex_or_bytes_to_address(single_booster_result[0:32])
+            convex_rewards_addrs.append(crv_rewards)
+            convex_lp_tokens_addrs.append(lp_token_addr)
+
+    lp_tokens_contract = EvmContract(  # only need it to encode and decode
+        address=ZERO_ADDRESS,
+        abi=ethereum.contracts.abi('CONVEX_LP_TOKEN'),
+        deployed_block=0,
+    )
+    calls_to_lp_tokens = [(lp_token_addr, lp_tokens_contract.encode('symbol')) for lp_token_addr in convex_lp_tokens_addrs]  # noqa: E501
+    lp_tokens_result = ethereum.multicall(calls=calls_to_lp_tokens)
+    queried_convex_pools_info: dict[ChecksumEvmAddress, str] = {}
+    for convex_reward_addr, single_lp_token_result in zip(convex_rewards_addrs, lp_tokens_result):
+        decoded_lp_token_symbol = lp_tokens_contract.decode(single_lp_token_result, 'symbol')[0]
+        queried_convex_pools_info[convex_reward_addr] = decoded_lp_token_symbol
```

Comment:
if `len(convex_lp_tokens_addrs)` you can exit early here just by returning `{}` right?

This avoids calling the contract and the multicall logic